### ISSUE TITLE
[BUG] Can't compile from source | error: Namespace Adw does not contain a class called Toast

### ISSUE THREAD

**Issue by mutageneral** [realmazharhussain/gdm-settings#16]
### What happened?

I followed the instruction to build the program, and it fails when I run `meson install -C build`. I also tried method 2. The error is something about `error: Namespace Adw does not contain a class called Toast`. I have downloaded all required dependencies with dnf, and I am on Fedora 35. This is my first time compiling from source, so I might have done mistakes.

### Expected behavior

For the application to compile correctly.

### To Reproduce

- download tar.gz and extract it
- Install all dependencies with dnf on fedora 35
- meson build
- meson install -C build

or

- Install all dependencies with dnf on fedora 35
- run method 2 blocks of commands

## Method 1 | meson build

[ordici@node-1w7jra94mv9v7n80n5tf3q5ok gdm-settings-main]$ meson build
The Meson build system
Version: 0.59.4
Source dir: /home/ordici/Documents/git/gdm-settings-main
Build dir: /home/ordici/Documents/git/gdm-settings-main/build
Build type: native build
Project name: gdm-settings
Project version: 0.3-alpha
Host machine cpu family: x86_64
Host machine cpu: x86_64
Found pkg-config: /usr/bin/pkg-config (1.8.0)
Run-time dependency pygobject-3.0 found: YES 3.42.0
Run-time dependency libadwaita-1 found: YES 1.0.0.alpha.4
Run-time dependency glib-2.0 found: YES 2.70.4
Program python3 found: YES (/usr/bin/python3)
Configuring application_id.appdata.xml using configuration
Program appstream-util found: YES (/usr/bin/appstream-util)
Configuring application_id.desktop using configuration
Program desktop-file-validate found: YES (/usr/bin/desktop-file-validate)
Configuring io.github.realmazharhussain.GdmSettings.gschema.xml using configuration
Configuring io.github.realmazharhussain.GdmSettings.appearance.gschema.xml using configuration
Configuring io.github.realmazharhussain.GdmSettings.fonts.gschema.xml using configuration
Configuring io.github.realmazharhussain.GdmSettings.misc.gschema.xml using configuration
Configuring io.github.realmazharhussain.GdmSettings.night-light.gschema.xml using configuration
Configuring io.github.realmazharhussain.GdmSettings.sound.gschema.xml using configuration
Configuring io.github.realmazharhussain.GdmSettings.top-bar.gschema.xml using configuration
Configuring io.github.realmazharhussain.GdmSettings.touchpad.gschema.xml using configuration
Configuring io.github.realmazharhussain.GdmSettings.tools.gschema.xml using configuration
Configuring io.github.realmazharhussain.GdmSettings.window-state.gschema.xml using configuration
Program glib-compile-schemas found: YES (/usr/bin/glib-compile-schemas)
Configuring info.py using configuration
Configuring gdm-settings using configuration
Program blueprint-compiler found: NO
Clonage dans 'blueprint-compiler'...
remote: Enumerating objects: 196, done.
remote: Counting objects: 100% (196/196), done.
remote: Compressing objects: 100% (152/152), done.
remote: Total 196 (delta 21), reused 155 (delta 12), pack-reused 0
Réception d'objets: 100% (196/196), 81.87 Kio | 1.34 Mio/s, fait.
Résolution des deltas: 100% (21/21), fait.

Executing subproject blueprint-compiler 

blueprint-compiler| Project name: blueprint-compiler
blueprint-compiler| Project version: 0.1.0
blueprint-compiler| Program python3 found: YES (/usr/bin/python3)
blueprint-compiler| Configuring blueprint-compiler.pc using configuration
blueprint-compiler| Program blueprint-compiler.py found: YES (/home/ordici/Documents/git/gdm-settings-main/subprojects/blueprint-compiler/blueprint-compiler.py)
blueprint-compiler| Build targets in project: 2
blueprint-compiler| Subproject blueprint-compiler finished.

Program blueprint-compiler.py found: YES (overridden)
Configuring main-window.blp using configuration
Program install found: YES (/usr/bin/install)
Configuring postinstall.py using configuration
Program postinstall.py found: YES (/home/ordici/Documents/git/gdm-settings-main/build/build-aux/meson/postinstall.py)
Build targets in project: 6

gdm-settings 0.3-alpha

  Subprojects
    blueprint-compiler: YES

Found ninja-1.10.2 at /usr/bin/ninja

## Method 1 | meson install -C build

[ordici@node-1w7jra94mv9v7n80n5tf3q5ok gdm-settings-main]$ meson install -C build
ninja: Entering directory `/home/ordici/Documents/git/gdm-settings-main/build'
[7/12] Generating 'src/blueprints/gdm-settings.p/main-window.ui'.
FAILED: src/blueprints/gdm-settings.p/main-window.ui 
/home/ordici/Documents/git/gdm-settings-main/subprojects/blueprint-compiler/blueprint-compiler.py compile src/blueprints/main-window.blp --output src/blueprints/gdm-settings.p/main-window.ui
error: Namespace Adw does not contain a class called ToastOverlay
at src/blueprints/main-window.blp line 29 column 9:
  29 |    Adw.ToastOverlay main_toast_overlay {
     |        ^
hint: Did you check your spelling?
hint: Are your dependencies up to date?

error: Namespace Adw does not contain a class called Toast
at src/blueprints/main-window.blp line 89 column 5:
  89 |Adw.Toast apply_failed_toast {
     |    ^
hint: Did you check your spelling?
hint: Are your dependencies up to date?

error: Namespace Adw does not contain a class called Toast
at src/blueprints/main-window.blp line 95 column 5:
  95 |Adw.Toast apply_succeeded_toast {
     |    ^
hint: Did you check your spelling?
hint: Are your dependencies up to date?

error: Namespace Adw does not contain a class called Toast
at src/blueprints/main-window.blp line 101 column 5:
 101 |Adw.Toast user_settings_imported_toast {
     |    ^
hint: Did you check your spelling?
hint: Are your dependencies up to date?

error: Namespace Adw does not contain a class called Toast
at src/blueprints/main-window.blp line 107 column 5:
 107 |Adw.Toast settings_reloaded_toast {
     |    ^
hint: Did you check your spelling?
hint: Are your dependencies up to date?

5 errors
[11/12] Generating 'src/blueprints/gdm-settings.p/tools-page.ui'.
ninja: build stopped: subcommand failed.
Could not rebuild /home/ordici/Documents/git/gdm-settings-main/build

## Method 2

[ordici@node-1w7jra94mv9v7n80n5tf3q5ok git]$ git clone --depth=1 https://github.com/realmazharhussain/gdm-settings
cd gdm-settings
meson build
meson install -C build
Clonage dans 'gdm-settings'...
remote: Enumerating objects: 94, done.
remote: Counting objects: 100% (94/94), done.
remote: Compressing objects: 100% (84/84), done.
remote: Total 94 (delta 11), reused 50 (delta 3), pack-reused 0
Réception d'objets: 100% (94/94), 609.30 Kio | 4.35 Mio/s, fait.
Résolution des deltas: 100% (11/11), fait.
The Meson build system
Version: 0.59.4
Source dir: /home/ordici/Documents/git/gdm-settings
Build dir: /home/ordici/Documents/git/gdm-settings/build
Build type: native build
Project name: gdm-settings
Project version: 0.3-alpha
Host machine cpu family: x86_64
Host machine cpu: x86_64
Found pkg-config: /usr/bin/pkg-config (1.8.0)
Run-time dependency pygobject-3.0 found: YES 3.42.0
Run-time dependency libadwaita-1 found: YES 1.0.0.alpha.4
Run-time dependency glib-2.0 found: YES 2.70.4
Program python3 found: YES (/usr/bin/python3)
Configuring application_id.appdata.xml using configuration
Program appstream-util found: YES (/usr/bin/appstream-util)
Configuring application_id.desktop using configuration
Program desktop-file-validate found: YES (/usr/bin/desktop-file-validate)
Configuring io.github.realmazharhussain.GdmSettings.gschema.xml using configuration
Configuring io.github.realmazharhussain.GdmSettings.appearance.gschema.xml using configuration
Configuring io.github.realmazharhussain.GdmSettings.fonts.gschema.xml using configuration
Configuring io.github.realmazharhussain.GdmSettings.misc.gschema.xml using configuration
Configuring io.github.realmazharhussain.GdmSettings.night-light.gschema.xml using configuration
Configuring io.github.realmazharhussain.GdmSettings.sound.gschema.xml using configuration
Configuring io.github.realmazharhussain.GdmSettings.top-bar.gschema.xml using configuration
Configuring io.github.realmazharhussain.GdmSettings.touchpad.gschema.xml using configuration
Configuring io.github.realmazharhussain.GdmSettings.tools.gschema.xml using configuration
Configuring io.github.realmazharhussain.GdmSettings.window-state.gschema.xml using configuration
Program glib-compile-schemas found: YES (/usr/bin/glib-compile-schemas)
Configuring info.py using configuration
Configuring gdm-settings using configuration
Program blueprint-compiler found: NO
Clonage dans 'blueprint-compiler'...
remote: Enumerating objects: 196, done.
remote: Counting objects: 100% (196/196), done.
remote: Compressing objects: 100% (152/152), done.
remote: Total 196 (delta 21), reused 155 (delta 12), pack-reused 0
Réception d'objets: 100% (196/196), 81.87 Kio | 9.10 Mio/s, fait.
Résolution des deltas: 100% (21/21), fait.

Executing subproject blueprint-compiler 

blueprint-compiler| Project name: blueprint-compiler
blueprint-compiler| Project version: 0.1.0
blueprint-compiler| Program python3 found: YES (/usr/bin/python3)
blueprint-compiler| Configuring blueprint-compiler.pc using configuration
blueprint-compiler| Program blueprint-compiler.py found: YES (/home/ordici/Documents/git/gdm-settings/subprojects/blueprint-compiler/blueprint-compiler.py)
blueprint-compiler| Build targets in project: 2
blueprint-compiler| Subproject blueprint-compiler finished.

Program blueprint-compiler.py found: YES (overridden)
Configuring main-window.blp using configuration
Program install found: YES (/usr/bin/install)
Configuring postinstall.py using configuration
Program postinstall.py found: YES (/home/ordici/Documents/git/gdm-settings/build/build-aux/meson/postinstall.py)
Build targets in project: 6

gdm-settings 0.3-alpha

  Subprojects
    blueprint-compiler: YES

Found ninja-1.10.2 at /usr/bin/ninja
ninja: Entering directory `/home/ordici/Documents/git/gdm-settings/build'
[4/12] Generating 'src/blueprints/gdm-settings.p/main-window.ui'.
FAILED: src/blueprints/gdm-settings.p/main-window.ui 
/home/ordici/Documents/git/gdm-settings/subprojects/blueprint-compiler/blueprint-compiler.py compile src/blueprints/main-window.blp --output src/blueprints/gdm-settings.p/main-window.ui
error: Namespace Adw does not contain a class called ToastOverlay
at src/blueprints/main-window.blp line 29 column 9:
  29 |    Adw.ToastOverlay main_toast_overlay {
     |        ^
hint: Did you check your spelling?
hint: Are your dependencies up to date?

error: Namespace Adw does not contain a class called Toast
at src/blueprints/main-window.blp line 89 column 5:
  89 |Adw.Toast apply_failed_toast {
     |    ^
hint: Did you check your spelling?
hint: Are your dependencies up to date?

error: Namespace Adw does not contain a class called Toast
at src/blueprints/main-window.blp line 95 column 5:
  95 |Adw.Toast apply_succeeded_toast {
     |    ^
hint: Did you check your spelling?
hint: Are your dependencies up to date?

error: Namespace Adw does not contain a class called Toast
at src/blueprints/main-window.blp line 101 column 5:
 101 |Adw.Toast user_settings_imported_toast {
     |    ^
hint: Did you check your spelling?
hint: Are your dependencies up to date?

error: Namespace Adw does not contain a class called Toast
at src/blueprints/main-window.blp line 107 column 5:
 107 |Adw.Toast settings_reloaded_toast {
     |    ^
hint: Did you check your spelling?
hint: Are your dependencies up to date?

5 errors
[9/12] Generating 'src/blueprints/gdm-settings.p/touchpad-page.ui'.
ninja: build stopped: subcommand failed.
Could not rebuild /home/ordici/Documents/git/gdm-settings/build

### Application Version

latest development version

### Source

Source Code

### Environment

```markdown
- **OS**: Fedora 35
- **GNOME**: 41.4
- *libadwaita-devel*: 1.0.0
- *glib-devel*: 1.2.10
- *python3-gobject-devel*: 3.42.0
```

**Comment by realmazharhussain** [comment]
Toast is one of the major features of LibAdwaita. It seems like some distributions have an older version of LibAdwaita in which this feature is missing. You are not doing anything wrong.

## Solution
Either you can wait for your distribution to update LibAdwaita to a newer version or you can install the latest version of LibAdwaita from source.

**Comment by enx1998** [comment]
Hi @realmazharhussain can you tell us what is the minimum version of LibAdwaita needed to compile the app, please? I'm trying to compile it on a gentoo box with LibAdwaita version 1.0.2 (the latest available on gentoo) and I have the same error.

Thanks ENx

**Comment by alesya-h** [comment]
Adw.ToolbarView is available only since adwaita 1.4, but meson.build states `dependency('libadwaita-1', version: '>=1.2')`. It must be updated.
